ext/redcarpet/buffer.c

```diff
@@ -74,7 +75,7 @@ bufgrow(struct buf *buf, size_t neosz)
 	while (neoasz < neosz)
 		neoasz += buf->unit;
 
-	neodata = realloc(buf->data, neoasz);
+	neodata = ruby_xrealloc(buf->data, neoasz);
 	if (!neodata)
```

Comment:
You can remove the existing NULL or no memory error checks now, since these functions will raise a no memory exception

Reply:
I've just seen that [the Rubinius build](https://travis-ci.org/vmg/redcarpet/jobs/300343417) has failed.

I don't have enough context here but it may be that those functions don't exist in Rubinius?

In that case, for backwards compatibility, maybe we should check `https://github.com/vmg/redcarpet/blob/f441dec42a5097530328b20e9d5ed1a025c600f7/ext/redcarpet/rc_markdown.c#L153` for being `NULL`, maybe raise a `NoMemoryError`?

Reply:
The build failed during `bundle install --without=benchmark`, so it didn't even get to `bundle exec rake` where it would compile the extension.  Given that there aren't any changes to Gemfile or redcarpet.gemspec, it doesn't look like this is related to that change.